### PR TITLE
[WIP] Initial work for getting testing setup with molecule

### PR DESCRIPTION
**Pull request by nigelbabu**
This is currently a work in progress. In the coming days, we'll add tests and instructions on how to run this and regular CI runs.

### CODE REVIEW

**Comment by nigelbabu** [comment]
I've also fixed any linting issues that this has thrown up.

**Comment by sabose** [comment]
@nigelbabu How can I try this out - could you add some documentation to run tests locally?

**Comment by nigelbabu** [comment]
@sabose I've added a testing section to the docs. It should probably not live there long-term, but I'm using that as a place to collect documentation for now.

File: roles/backend_setup/README.md [hunk 1]
```diff
@@ -186,6 +186,55 @@ Configure the ports and services related to GlusterFS, create logical volumes an
...
+        $ sudo groupadd docker
+        $ sudo usermod -aG docker $USER
+
+* Verify everything works with the following:
+
+        $ docker run hello-world
+
+* If you have SELinux enabled, run the following
+
+        $ sudo yum install libselinux-python
```
Comment (sabose) [inline roles/backend_setup/README.md]: Also sudo yum install gcc python-virtualenv

File: roles/backend_setup/README.md [hunk 2]
```diff
@@ -186,6 +186,55 @@ Configure the ports and services related to GlusterFS, create logical volumes an
...
+
+* Create a virtualenv
+
+        $ virtualenv --system-site-packaes env
```
Comment (sabose) [inline roles/backend_setup/README.md]: minor nit --system-site-packages

Also add a step to activate the env (prior to pip install step)
$ source env/bin/activate

**Comment by nigelbabu** [comment]
Looks like Sheersha's patch which does this better is now merged. I'm going to close this one.

**Pull request closed without merging**